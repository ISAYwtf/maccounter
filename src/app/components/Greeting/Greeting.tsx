import React from 'react'
// import styles from './Greeting.module.css'

interface GreetingProps {
    userName: string,
}

const Greeting: React.FC<GreetingProps> = ({ userName }) => (
    <div className="txt--title-small">
        <p>
            Good morning,&nbsp;
            <span>{userName}</span>
            !
        </p>
    </div>
)

export default Greeting
