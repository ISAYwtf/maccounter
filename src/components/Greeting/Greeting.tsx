import React from 'react'
// import styles from './Greeting.module.css'

const Greeting: React.FC = () => (
    <div className="txt--title-small">
        <p>
            Good morning,
            <span>User</span>
            !
        </p>
    </div>
)

export default Greeting
