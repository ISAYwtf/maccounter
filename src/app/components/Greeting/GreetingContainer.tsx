import Greeting from '@components/Greeting/Greeting'
import { setTimePeriod } from '@store/app-store/appReducer'
import { useAppSelector } from '@store/hooks'
import React from 'react'
import { useDispatch } from 'react-redux'

const GreetingContainer: React.FC = () => {
    const dispatch = useDispatch()
    const userName = useAppSelector(((state) => state.profile.firstName))
    const timePeriod = useAppSelector((state) => state.app.timePeriod)
    dispatch(setTimePeriod())
    return <Greeting userName={userName} timePeriod={timePeriod} />
}

export default GreetingContainer
