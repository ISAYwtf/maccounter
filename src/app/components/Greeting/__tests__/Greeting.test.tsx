import { render } from '@testing-library/react'
import React from 'react'
import Greeting from '../Greeting'

describe('Greeting', () => {
    const userName = 'Jack'
    const renderComponent = <Greeting userName={userName} />

    it('render Greeting', () => {
        const { asFragment } = render(renderComponent)
        expect(asFragment()).toMatchSnapshot()
    })
})
