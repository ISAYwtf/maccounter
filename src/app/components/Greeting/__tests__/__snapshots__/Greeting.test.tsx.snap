// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Greeting render Greeting 1`] = `
<DocumentFragment>
  <div
    class="txt--title-small"
  >
    <p>
      Good morning,Â 
      <span>
        Jack
      </span>
      !
    </p>
  </div>
</DocumentFragment>
`;
